owncloudApp/src/main/java/com/owncloud/android/presentation/ui/security/PatternActivity.kt

```diff
@@ -0,0 +1,322 @@
+                     */
+                    if (patternValue.isNullOrEmpty()) {
+                        patternValue = PatternLockUtils.patternToString(patternLockView, list)
+                    } else {
+                        newPatternValue = PatternLockUtils.patternToString(patternLockView, list)
+                    }
+                } else {
+                    patternValue = PatternLockUtils.patternToString(patternLockView, list)
+                }
+                Timber.d("Pattern %s", PatternLockUtils.patternToString(patternLockView, list))
+                processPattern()
+            }
+
+            override fun onCleared() {
+                Timber.d("Pattern has been cleared")
+            }
+        })
+    }
+
+    private fun processPattern() {
+        if (ACTION_CHECK == intent.action) {
+            /**
+             * This block is executed when the user opens the app after setting the pattern lock
+             * this block takes the pattern input by the user and checks it with the pattern initially set by the user.
+             */
+            if (patternViewModel.checkPatternIsValid(patternValue)) {
+                patternError.visibility = View.INVISIBLE
+                val preferencesProvider = SharedPreferencesProviderImpl(applicationContext)
+                preferencesProvider.putLong(PREFERENCE_LAST_UNLOCK_TIMESTAMP, System.currentTimeMillis())
+                finish()
+            } else {
+                showErrorAndRestart(
+                    R.string.pattern_incorrect_pattern,
+                    R.string.pattern_enter_pattern, View.INVISIBLE
+                )
+            }
+        } else if (ACTION_CHECK_WITH_RESULT == intent.action) {
+            //This block is executed when the user is removing the pattern lock (i.e disabling the pattern lock)
+            if (patternViewModel.checkPatternIsValid(patternValue)) {
+                patternViewModel.removePattern()
+                val result = Intent()
+                setResult(RESULT_OK, result)
+                patternError.visibility = View.INVISIBLE
+                notifyDocumentProviderRoots(applicationContext)
+                finish()
+            } else {
+                showErrorAndRestart(
+                    R.string.pattern_incorrect_pattern,
+                    R.string.pattern_enter_pattern, View.INVISIBLE
+                )
+            }
+        } else if (ACTION_REQUEST_WITH_RESULT == intent.action) {
+            //This block is executed when the user is setting the pattern lock (i.e enabling the pattern lock)
+            if (!confirmingPattern) {
+                patternError.visibility = View.INVISIBLE
+                requestPatternConfirmation()
+            } else if (confirmPattern()) {
+                savePatternAndExit()
+            } else {
+                showErrorAndRestart(
+                    R.string.pattern_not_same_pattern,
+                    R.string.pattern_enter_pattern, View.VISIBLE
```

Comment:
When we have 3 or more parameters, I would recommend using named params. It will make the code easier to read. For example, at the moment I don't know what `showErrorAndRestart` does, but if it had named params, I could imagine 🤔